test/utils.js

```diff
@@ -36,11 +36,28 @@ export function almostEqual(a, b, criteria) {
   }
 }
 
-export function checkValue(
-    output, expected, criteria = opFp32AccuracyCriteria) {
+export function compareUlp(a, b, nulp = 1, base = 64) {
+  const aBuffer = new ArrayBuffer(8);
+  const bBuffer = new ArrayBuffer(8);
+  const aView = new DataView(aBuffer);
+  const bView = new DataView(bBuffer);
+
+  if (base === 64) {
+    aView.setFloat64(0, a, true);
+    bView.setFloat64(0, b, true);
+  } else if (base === 32) {
+    aView.setFloat32(0, a, true);
+    bView.setFloat32(0, b, true);
+  }
+
+  return Math.abs(new BigUint64Array(aBuffer) - new BigUint64Array(bBuffer)) >
```

Comment:
For `getBitwise`, you could return `aView.getBigInt64(0)` here.

Reply:
`Math.abs` doesn't work with `BigInt64`, so you probably need to implement it by yourself. Fortunately, it would straightfoward, as BigInt64 support comparison and negate.